source/nodejs/adaptivecards/src/carousel.ts

```diff
@@ -417,6 +450,41 @@ export class Carousel extends Container {
             }
         );
     }
+
+    private createCarouselEvent (type : Enums.CarouselInteractionEvent): CarouselEvent
+    {
+        let currentPageId : string | undefined;
+        if (this.currentPageIndex != undefined) {
+            currentPageId = this.getItemAt(this.currentPageIndex).id;
+        }
+        return new CarouselEvent(type, this.id, currentPageId, this.currentPageIndex);   
+    }
+
+    private raiseCarouselEvent(eventType : Enums.CarouselInteractionEvent) {
+        const onCarouselEventHandler = AdaptiveCard.onCarouselEvent;
+        if (onCarouselEventHandler && eventType == Enums.CarouselInteractionEvent.Pagination) {
+            onCarouselEventHandler(this.createCarouselEvent(this.currentEventType));
+        }
+        this.currentEventType = eventType;
+    }
+}
+
+export class CarouselEvent {
+    type : Enums.CarouselInteractionEvent;
+    carouselId : string | undefined;
+    activeCarouselPageId : string | undefined;
+    activeCarouselPageIndex : number | undefined;
+
+    constructor(type : Enums.CarouselInteractionEvent,
+        carouselId : string | undefined,
+        activeCarouselPageId : string | undefined,
+        activeCarouselPageIndex : number | undefined)
+    {
+        this.carouselId = carouselId;
```

Comment:
nit: 
typescript has simple syntax which doesn't need you to have `this.carouselId = carouselId;` and `type : Enums.CarouselInteractionEvent;`

just declare public in constructor
```
constructor(public type : Enums.CarouselInteractionEvent...) {}
```

![image](https://user-images.githubusercontent.com/6290692/178656376-6c420afa-46a1-47f4-bf9c-f3da174e1bd5.png)